azureml/components/src/jsonl_knn_cosine_similarity.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import functools
+import json
+import pathlib
+
+import numpy as np
+
+
+from shared.jsonl_file_utils import load_jsonl
+from shared.jsonl_utils import line_map
+from shared.logging_utils import get_standard_logger_for_file
+
+
+_logger = get_standard_logger_for_file(__file__)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(add_help=True)
+
+    # Information about the datasets
+    datasets_group = parser.add_argument_group("Datasets")
+    datasets_group.add_argument("--input_dataset", type=pathlib.Path, required=True)
+    datasets_group.add_argument("--input_encoding", type=str, required=True)
+    datasets_group.add_argument("--output_dataset", type=pathlib.Path, required=True)
+    datasets_group.add_argument("--output_encoding", type=str, required=True)
+    datasets_group.add_argument("--example_dataset", type=pathlib.Path, required=True)
+    datasets_group.add_argument("--example_encoding", type=str, required=True)
+
+    # Information about keys
+    key_group = parser.add_argument_group("Keys")
+    key_group.add_argument("--input_vector_key", type=str, required=True)
+    key_group.add_argument("--example_vector_key", type=str, required=True)
+    key_group.add_argument("--output_key", type=str, required=True)
+
+    # Information about the algorithm
+    algo_group = parser.add_argument_group("Algorithm")
+    algo_group.add_argument("--k_nearest", type=int, required=True)
+
+    args = parser.parse_args()
+    return args
+
+
+def compute_knn(
+    item: dict[str, any],
+    *,
+    examples: list[dict[str, any]],
+    example_embedding_matrix: np.ndarray,
+    input_vector_key: str,
+    output_key: str,
+    k_nearest: int,
+) -> dict[str, any]:
+    # _logger.info(f"process_item: {item}")
+
+    item_embedding = np.asarray(item[input_vector_key])
```

Comment:
@Harsha-Nori I'm concerned that this implementation is overly simplistic when it comes to computing the k-nearest-neighbours. I believe I saw small amounts of randomness being added in the command line implementation?